proc-macros/src/rpc_macro.rs

```diff
@@ -57,10 +57,7 @@ impl RpcMethod {
 		let aliases = parse_aliases(aliases)?;
 		let blocking = optional(blocking, Argument::flag)?.is_some();
 		let name = name?.string()?;
-		let param_format = match param_format {
-			Ok(param_format) => param_format.string()?,
-			Err(_) => String::from("array"),
-		};
+		let param_format = optional(param_format, Argument::string)?.or(Some("array".into()));
```

Comment:
sorry if my suggestion was bad, but would be better to check `array` or `map` here then to introduce another type `Params` instead of `String`.

Such as:

```rust

enum Params {
   Array,
   Map,
}

let param_format = match optional(param_format, Argument::string)? {
  None | Some("array") => Params::Array,
  Some("map") => Params::Map,
  err => panic!("param_format must be either `map` or `array`, got {:?}", err),
};
```

Reply:
this could be perhaps be a separate function as `fn parse_param_format` or something?!

to avoid duplicating the code...

Reply:
Ah, now I get you. I wasn't sure whether you would embrace an additional type, but I agree, it makes things much cleaner.

Reply:
> this could be perhaps be a separate function as fn parse_param_format or something?!

Agreed.